hera_mc/cm_hookup.py

```diff
@@ -537,7 +508,7 @@ def _get_connection(self, current):
         options = list(self.active.connections[current.direction][current.key].keys())
         try:
             current.type = self.active.parts[current.key].hptype
-        except KeyError:
+        except KeyError:  # pragma: no cover
```

Comment:
What is the use case where this can happen? Why not test it?

Reply:
There are a couple of places I have this for the following reason:  the way this works now is via the get_active stuff, which disallows it from happening.  I could see someone (maybe) wanting to use this from a different call and I wanted to retain the check, but not put in the extra test infrastructure (a whole other module) to test.

Reply:
Ok. I can't tell you how many times I've discovered that my error handling code doesn't work as expected (either doesn't get invoked when I expected it to or doesn't handle the problem as expected) until I add tests that actually exercise it.

Reply:
I don't disagree, and I've contorted to check most stuff.  The other option is to remove the exception and let it error.  Since it is operational, it seemed better to catch it.